feature/security/gnsi/credentialz/tests/hiba_authentication/hiba_authentication_test.go

```diff
@@ -0,0 +1,690 @@
+							JsonIetfVal: roleData,
+						},
+					},
+				},
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "system"},
+							{Name: "aaa"},
+							{Name: "authentication"},
+							{Name: "user", Key: map[string]string{"username": username}},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: userData,
+						},
+					},
+				},
+			},
+		}
+		gnmiClient := dut.RawAPIs().GNMI(t)
+		if _, err = gnmiClient.Set(context.Background(), SetRequest); err != nil {
+			t.Fatalf("Unexpected error configuring User: %v", err)
+		}
+	default:
+		t.Fatalf("Unsupported vendor %s for deviation 'deviation_native_users'", dut.Vendor())
+	}
+}
+
+func setupUser(t *testing.T, dut *ondatra.DUTDevice) {
+	auth := &oc.System_Aaa_Authentication{}
+	user := auth.GetOrCreateUser(username)
+
+	if deviations.SetNativeUser(dut) {
+		// probably all vendors need to handle this since the user should have a role attached to
+		// it allowing us to login via ssh/console/whatever
+		createNativeRole(t, dut, "credz-fp-test")
+	} else {
+		user.SetPassword(password)
+	}
+
+	gnmi.Update(t, dut, gnmi.OC().System().Aaa().Authentication().Config(), auth)
+
+}
+
+func setupDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	// set only pub key auth for our test
+	setupAuthenticationTypes(t, dut)
+
+	// set the authorized principals thing
+	setupAuthorizedPrincipals(t, dut)
+}
+
+func sendHostParametersRequest(t *testing.T, dut *ondatra.DUTDevice, request *credentialz.RotateHostParametersRequest) {
+	credzClient := dut.RawAPIs().GNSI(t).Credentialz()
+
+	credzRotateClient, err := credzClient.RotateHostParameters(context.Background())
+	if err != nil {
+		t.Fatalf("failed fetching credentialz rotate host parameters client, error: %s", err)
+	}
+
+	err = credzRotateClient.Send(request)
+	if err != nil {
+		t.Fatalf("failed sending credentialz rotate host parameters request, error: %s", err)
+	}
+
+	_, err = credzRotateClient.Recv()
+	if err != nil {
+		t.Fatalf("failed receiving credentialz rotate host parameters request, error: %s", err)
+	}
+
+	err = credzRotateClient.Send(&credentialz.RotateHostParametersRequest{
+		Request: &credentialz.RotateHostParametersRequest_Finalize{
+			Finalize: request.GetFinalize(),
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed sending credentialz rotate host parameters finalize request, error: %s", err)
+	}
+}
+
+func setupAuthenticationTypes(t *testing.T, dut *ondatra.DUTDevice) {
+	request := &credentialz.RotateHostParametersRequest{
+		Request: &credentialz.RotateHostParametersRequest_AuthenticationAllowed{
+			AuthenticationAllowed: &credentialz.AllowedAuthenticationRequest{
+				AuthenticationTypes: []credentialz.AuthenticationType{
+					credentialz.AuthenticationType_AUTHENTICATION_TYPE_PUBKEY,
+				},
+			},
+		},
+	}
+
+	sendHostParametersRequest(t, dut, request)
+}
+
+func setupAuthorizedPrincipals(t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
Not keen on the name of this function, as it is "turning on hiba for AuthorizedPrincipalCommand" rather than setting AuthorizedPrincipals (which at first read I'd take to mean setting principals in an authorizedprincipalsfile.

Reply:
sure, is `enableAuthorizedPrincipals()` more satisfactory? If you had something in mind please let me know and I'll change it.

Reply:
Updated the name to `setupAuthorizedPrincipalsCommand`